src/phase0/state_transition/mod.rs

```diff
@@ -60,6 +60,18 @@ pub enum Error {
         previous: Epoch,
         current: Epoch,
     },
+    #[error("invalid block header: {0}")]
+    InvalidHeader(InvalidHeader),
+}
+
+#[derive(Debug, Error)]
+pub enum InvalidHeader {
+    #[error("slots mismatch")]
+    StateSlotMismatch,
```

Comment:
i'd suggest we put as much information into the error as we have from the context in which it is created:


```suggestion
    #[error("mismatch between state slot {state_slot} and block slot {block_slot}")]
    StateSlotMismatch { state_slot: Slot, block_slot: Slot },
```